internal/converter/types.go

```diff
@@ -61,11 +62,10 @@ componentLoop:
 }
 
 // Convert a proto "field" (essentially a type-switch with some recursion):
-func (c *Converter) convertField(curPkg *ProtoPackage, desc *descriptor.FieldDescriptorProto, msg *descriptor.DescriptorProto) (*jsonschema.Type, error) {
-
+func (c *Converter) convertField(curPkg *ProtoPackage, desc *descriptor.FieldDescriptorProto, msg *descriptor.DescriptorProto, duplicatedMessages map[*descriptor.DescriptorProto]string) (*jsonschema.Type, error) {
 	// Prepare a new jsonschema.Type for our eventual return value:
 	jsonSchemaType := &jsonschema.Type{
-		Properties: make(map[string]*jsonschema.Type),
+		Properties: orderedmap.New(),
```

Comment:
If you were to remove this line then we would no longer have the empty `"properties": {},` cruft in the generated schemas.

How do you feel about that? I think it is a good idea to keep the generated schemas as clean as possible. You'd have to clean up the schemas in the unit tests though, which I know is a horrible job.